client/redis.go

```diff
@@ -530,12 +544,16 @@ func DialWithConfig(cfg *DialConfig) (*Redis, error) {
 	if cfg.MaxIdle == 0 {
 		cfg.MaxIdle = DefaultMaxIdle
 	}
+	if cfg.TCPKeepAlive == 0 {
```

Comment:
How do we handle `0` provided by the user? It is going to be always changed to the default here.
How about a pointer and a nil check?